support/k8s/k8s-workload-registrar/informerhandler.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+)
+
+type InformerHandlerConfig struct {
+	Log        logrus.FieldLogger
+	Controller *Controller
+	Factory    informers.SharedInformerFactory
+}
+
+type InformerHandler struct {
+	c InformerHandlerConfig
+}
+
+func NewInformerHandler(config InformerHandlerConfig) *InformerHandler {
+	return &InformerHandler{
+		c: config,
+	}
+}
+
+func (ih *InformerHandler) Run(ctx context.Context) error {
+	podInformer := ih.c.Factory.Core().V1().Pods().Informer()
+	podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+
+		AddFunc: func(obj interface{}) {
+			pod := obj.(*corev1.Pod)
+			log := ih.c.Log.WithFields(logrus.Fields{
+				"ns":  pod.Namespace,
+				"pod": pod.Name,
+			})
+			if err := ih.c.Controller.SyncPod(ctx, pod); err != nil {
+				log.WithError(err).Error("Failed to sync pod")
+			}
+		},
+
+		UpdateFunc: func(old, new interface{}) {
```

Comment:
What's the difference between this and AddFunc? Seems like `SyncPod` already tries to be idempotent. Maybe these funcs can be defined on their own and we can load up this struct in a more succinct way, using the same func for Add and Update?

Reply:
Actually redundant after what I then did in SyncPod. I'll tidy up.